src/Http/RequestUtil.php

```diff
@@ -16,11 +16,23 @@ class RequestUtil
 {
     public static function getActor(Request $request): User
     {
-        return $request->getAttribute('actor');
+        return $request->getAttribute('actorReference')->getActor();
     }
 
     public static function withActor(Request $request, User $actor): Request
     {
-        return $request->withAttribute('actor', $actor);
+        $actorReference = $request->getAttribute('actorReference');
+
+        if (! $actorReference) {
+            $actorReference = new ActorReference;
+            $request = $request->withAttribute('actorReference', $actorReference);
+        }
```

Comment:
Not a fan of this at all, but I preferred to go with this instead of explicitly setting the actor reference in `Api\Client`

Reply:
Why not? If the whole goal of these methods is to encapsulate the request's user, the actual mechanism used for storing the user should be an implementation detail of the methods.

> explicitly setting the actor reference in Api\Client

We definitely wouldn't want this, it'd break encapsulation.